### PR TITLE
standard-OCL: switch to CLVideoBuffer path

### PR DESCRIPTION
**Pull request by liuyinhangx**
* build: download cl_intel.h from v1.3.0 release branch
* standard-OCL: fix invalid length for vector component access
* standard-OCL: fix row pitch aligned issue
* standard-OCL: switch to CLVideoBuffer path

### CODE REVIEW

**Comment by liuyinhangx** [comment]
Update patch (standard-OCL: switch to CLVideoBuffer path):
  xcamfilter: CLVideoBuffer support only CPU copy mode, add a check for the copy mode.

File: modules/ocl/cl_utils.cpp [hunk 1]
```diff
@@ -86,7 +86,19 @@ convert_to_climage (
...
+        if (offset == 0) {
+            main_buf = cl_buf;
+        } else {
+            uint32_t row_pitch = CLImage::calculate_pixel_bytes (desc.format) *
+                                     XCAM_ALIGN_UP (desc.width, XCAM_CL_IMAGE_ALIGNMENT_X);
+            uint32_t size = row_pitch * desc.height;
+
+            main_buf = new CLSubBuffer (context, cl_buf, flags, offset, size);
+        }
+
```
Comment (windyuan) [inline modules/ocl/cl_utils.cpp]: get_cl_buffer returned a cl_buffer(main_buf) and lost the buffer proxy(cl_video_buf) which may release the buffer to pool. but the cl_buf still in use in Image2D.

**Comment by liuyinhangx** [comment]
Update patch, please help to review, thanks.

**Comment by windyuan** [comment]
merged. thanks

commit cff638b59385036cbc91873a9965282c8861a306
commit 9e7e0ab2f59470fae36f28999520e01cb3a2299f
commit 86b01deaf21f20c19b5e069e0a3701ba0b50bbce
commit 224abf27cb56202f1da9c9cb7b93aedec822b61a

**Comment by windyuan** [comment]
changed set_buf_size to proteced.
	
    diff --git a/modules/ocl/cl_memory.h b/modules/ocl/cl_memory.h
	index c38ac1b..9faba88 100644
	--- a/modules/ocl/cl_memory.h
	+++ b/modules/ocl/cl_memory.h
	@@ -115,12 +115,14 @@ public:
			 CLEventList &event_waits = CLEvent::EmptyList,
			 SmartPtr<CLEvent> &event_out = CLEvent::NullEvent);

	+    uint32_t get_buf_size () const {
	+        return _size;
	+    }
	+
	+protected:
		 void set_buf_size (uint32_t size) {
			 _size = size;
		 }
	-    uint32_t get_buf_size () {
	-        return _size;
	-    }

**Pull request closed without merging**